libs/constrained-generators/src/Constrained/Spec/Tree.hs

```diff
@@ -93,29 +94,29 @@ instance HasSpec fn a => HasSpec fn (BinTree a) where
       : (BinNode left a <$> shrinkWithTypeSpec s right)
       ++ ((\l -> BinNode l a right) <$> shrinkWithTypeSpec s left)
 
-  toPreds t (BinTreeSpec sz s) =
+  toPreds t (BinTreeSpec msz s) =
     (forAll t $ \n -> n `satisfies` s)
-      <> genHint sz t
+      <> maybe TruePred (flip genHint t) msz
```

Comment:
What's the idea here? Why do we only give the generator a hint if the size is given in the spec and set it to `TrueSpec` otherwise?

Reply:
The idea is that because `genHint` is optional we only want to have it in the predicates if the user has explicitly given one.